#! /bin/sh /usr/share/dpatch/dpatch-run
## 08-escaping-caracters.dpatch by  <gilir@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: upstream bug http://bugzilla.gnome.org/show_bug.cgi?id=529301
## CP: ubuntu bug https://bugs.launchpad.net/ubuntu/+source/conduit/+bug/297085

@DPATCH@
diff -urNad conduit~/conduit/platform/FileGio.py conduit/conduit/platform/FileGio.py
--- conduit~/conduit/platform/FileGio.py	2009-08-30 13:33:34.000000000 +0200
+++ conduit/conduit/platform/FileGio.py	2009-08-30 14:06:49.000000000 +0200
@@ -1,4 +1,5 @@
 import gio
+import urllib
 
 import conduit.platform
 
@@ -200,6 +201,8 @@
         #if not relative, return abs path
         if not res:
             res = toURI
+        else:
+            res = urllib.quote(res)
         return res
 
     @staticmethod
@@ -353,10 +356,10 @@
                     if filetype == gio.FILE_TYPE_DIRECTORY:
                         #Include hidden directories
                         if not hidden or self.includeHidden:
-                            self.dirs.append(dir+"/"+filename)
+                            self.dirs.append(dir+"/"+urllib.quote(filename))
                             t += 1
                     elif filetype == gio.FILE_TYPE_REGULAR or (filetype == gio.FILE_TYPE_SYMBOLIC_LINK and self.followSymlinks):
-                            uri = dir+"/"+filename
+                            uri = dir+"/"+urllib.quote(filename)
                             #Include hidden files
                             if not hidden or self.includeHidden:
                                 self.URIs.append(uri)
